spec/features/ideas/user_can_create_an_idea_spec.rb

```diff
@@ -3,6 +3,9 @@
 describe 'User visits the new idea path to create a new idea' do
   describe 'They can fill in details for an idea and click submit' do
     scenario 'They can see that new idea\'s details on the ideas index page' do
+      user = User.create!(username: 'Nikhil', password: 'password')
+      allow_any_instance_of(ApplicationController).
```

Comment:
@AliSchlereth @iandouglas I worked on the Ideas Controller to use current_user and now each of my users has unique ideas. So that part is great. The issue now is that my tests that were passing earlier are now failing. I added the user and also code to make the user a current_user for the test. But for some reason when it reaches the click_on 'Create Idea' line, it gives a No route matches error. However, if I run this code via rails s and localhost, the code works great with no issues in create, editing, deleting an idea or ideas, that are specific to a user. I suspect I am missing some setup etc in my test. Would love your thoughts on what I could be missing (as the code is working).

Reply:
@Shniks Looking at your ideas new form and your routes, I'm surprised it is working on localhost. Your form is built for [@category, @idea] which suggests a route of category_ideas_path, but you don't have nested routes. Since you have a category drop down and a user can select a category when they create an idea, I don't see the need for nested routes.